### ISSUE TITLE
Texture generates "Inf" compression scalers values

### ISSUE THREAD

**Issue by hybridherbst** [needle-tools/procedural-stochastic-texturing#6]
Hi again @thomasdeliot,

I found a texture that generates infinity values and (currently, I have a fix) breaks the shader generation.
Are infinity values "by design" or a bug?

The texture:
![aerial_rocks_01_rough_1k](https://user-images.githubusercontent.com/2693840/105360615-14454380-5bf9-11eb-8482-d77f1f2aa1ed.jpg)

Generated data (in debug mode) - note infinity values in compression scalers:
![image](https://user-images.githubusercontent.com/2693840/105360705-2c1cc780-5bf9-11eb-8ba7-395e90d509cf.png)

I assume this is because generating a color space basis for a greyscale image currently fails?

**Comment by ghost** [comment]
Yes, the "Color" mode of ProceduralTexture2D assumes a three channel input where it can find a decorrelated colour space through PCA. Ideally one should use the "Other" mode ofr grayscale input :)

But in any case, using Color mode with grayscale input shouldn't fail. The PCA will only find one valid vector, and the two others will result in infinity compression scalers due to the division in ComputeCompressionScalers(), but the sampling will still work in the shader as in the end we'll only use the valid principal channel values.

I don't know why passing Inf values breaks the shader generation now though, that's weird ! The simple fix is to set the compression to 1 if the vector magnitude is null, instead of computing the division. Probably the fix you were thinking of already :)
